pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -1151,3 +1172,40 @@ func hasContainerInCrashloop(pod *corev1.Pod) bool {
 	}
 	return false
 }
+
+// LineLimitReader returns a Reader that reads from r
+// but stops with EOF after n lines.
+// The underlying implementation is a *LineLimitedReader.
+func LineLimitReader(r io.Reader, n int64) io.Reader { return &LineLimitedReader{r, n} }
+
+// A LineLimitedReader reads from R but limits the amount of
+// data returned to just N lines. Each call to Read
+// updates N to reflect the new amount remaining.
+// Read returns EOF when N <= 0 or when the underlying R returns EOF.
+type LineLimitedReader struct {
+	R io.Reader // underlying reader
```

Comment:
do not use single letter and **exported** struct vars... name this `reader` and the below `maxLinesLimit`... var names are cheap and good for reading.